File: Python/Sum of Left Leaves.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def sumOfLeftLeaves(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        self.sum = 0
        self.find(root)
        return self.sum
        
    def is_leaf(self, node):
        return node.left == None and node.right == None
    
    def find(self, root):
        if root is None:
            return
        if root.left != None and self.is_leaf(root.left):
            self.sum += root.left.val
        else:
            self.find(root.left)
        self.find(root.right)

File: cpp/Sum of Left Leaves.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int sum = 0;
    int sumOfLeftLeaves(TreeNode* root) {
        find(root);
        return sum;
    }

    bool is_leaf(TreeNode* node) {
        return node->left == NULL && node->right == NULL;
    }

    void find(TreeNode* root) {
        if (!root) {
            return;
        }
        if (root->left != NULL && is_leaf(root->left)) {
            sum += root->left->val;
        } else {
            find(root->left);
        }
        find(root->right);
    }
};
